python/hsfs/engine/spark.py

```diff
@@ -165,26 +167,43 @@ def save_dataframe(
         online_write_options,
         validation_id=None,
     ):
-
-        if storage == "offline" or not online_enabled:
-            self._save_offline_dataframe(
-                feature_group,
-                dataframe,
-                operation,
-                offline_write_options,
-                validation_id,
-            )
-        elif storage == "online":
-            self._save_online_dataframe(feature_group, dataframe, online_write_options)
-        elif online_enabled and storage is None:
-            self._save_offline_dataframe(
-                feature_group,
-                dataframe,
-                operation,
-                offline_write_options,
-            )
-            self._save_online_dataframe(feature_group, dataframe, online_write_options)
-        else:
+        try:
+            if isinstance(feature_group, StreamFeatureGroup):
+                self._save_online_dataframe(
```

Comment:
I think it is not possible to write to online storage in python client right? Need to throw exception in hive.py?

Reply:
yeah we should have all methods that are not implemented in hive.py throw proper exceptions, I have a Jira for it. So we can do it all at once, doesn't need to be in this PR